src/components/ContentNode/TabNavigator.vue

```diff
@@ -0,0 +1,111 @@
+<!--
+  This source file is part of the Swift.org open source project
+
+  Copyright (c) 2021 Apple Inc. and the Swift project authors
+  Licensed under Apache License v2.0 with Runtime Library Exception
+
+  See https://swift.org/LICENSE.txt for license information
+  See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+-->
+
+<template>
+  <div
+    class="TabNavigator"
+    :class="[{ 'tabs--vertical': vertical }]"
+  >
+    <Tabnav v-model="current" v-bind="{ position, vertical }">
+      <TabnavItem v-for="(item, index) in items" :key="item.title" :value="index">
+        {{ item.title }}
+      </TabnavItem>
+    </Tabnav>
+    <div class="tabs-content">
+      <ContentNode :content="currentContent" />
+    </div>
+  </div>
+</template>
+
+<script>
+import Tabnav from 'docc-render/components/Tabnav.vue';
+import TabnavItem from 'docc-render/components/TabnavItem.vue';
+
+/**
+ * Tab navigation component, that renders `ContentNode`,
+ * with ability to align the tabs horizontally and vertically to the start/center/end of the line.
+ * It can also flip the navigation, so it renders content first, navigation second,
+ * in both horizontal and vertical mode.
+ */
+export default {
+  name: 'TabNavigator',
+  components: {
+    TabnavItem,
+    Tabnav,
+    ContentNode: () => import('docc-render/components/ContentNode.vue'),
```

Comment:
I experimented a bit with this and came up with a way of doing this without the recursive import. The main benefit is that it removes the explicit `ContentNode` dependency and allows the content to be anything. It does utilize the `renderChildren` and slots in `ContentNode` as we described.

The commit/diff can be seen here: https://github.com/dobromir-hristov/swift-docc-render/compare/dhristov/97714987/add-tabnavigator-support...mportiz08:swift-docc-render:experiment/use-slots-for-tabnavigator

 You're welcome to try it out and merge it if you like it better (would need to update tests). Also, no worries if you don't think that method is better and prefer how it is already working here. I mostly just wanted to try and see if it would work.

Reply:
I looked at it and honestly I would not have thought of inverting it like that and having a slot for every tab title 🤣 That is deff an interesting approach.

Only potential downside I can think of now is that we are making Vue create all those slots but conditionally render only some.

Upside is I guess we can reuse the component in the future, without having to provide a `contentNode` compatible slot content.

I am OK with using it, your call :)

Reply:
Your call, it's your PR 😛